block/local/adapter_test.go

```diff
@@ -141,3 +142,56 @@ func TestLocalCopy(t *testing.T) {
 		t.Errorf("expected to read \"%s\" as written, got \"%s\"", contents, string(got))
 	}
 }
+
+func TestRemove(t *testing.T) {
+	a := makeAdapter(t)
+	const contents = "foo bar baz quux"
+
+	obj1 := makePointer("README")
+	err := a.Put(obj1, 0, strings.NewReader(contents), block.PutOpts{})
+	testutil.MustDo(t, "Put", err)
+	err = a.Remove(obj1)
+	testutil.MustDo(t, "Remove", err)
+
+	root := a.Path()
+	tree1 := dumpPathTree(t, root)
+	if diff := deep.Equal(tree1, []string{""}); diff != nil {
+		t.Fatalf("Found diff after remove: %s", diff)
+	}
+
+	obj2 := makePointer("src/tools.go")
+	err = a.Put(obj2, 0, strings.NewReader(contents), block.PutOpts{})
+	testutil.MustDo(t, "Put", err)
+	err = a.Remove(obj2)
+	testutil.MustDo(t, "Remove", err)
+	tree2 := dumpPathTree(t, root)
+	if diff := deep.Equal(tree2, []string{""}); diff != nil {
+		t.Fatalf("Found diff after remove: %s", diff)
+	}
+
+	obj3 := makePointer("a/b/c/d.txt")
+	err = a.Put(obj3, 0, strings.NewReader(contents), block.PutOpts{})
+	testutil.MustDo(t, "Put", err)
+	err = a.Remove(obj3)
+	testutil.MustDo(t, "Remove", err)
+	tree3 := dumpPathTree(t, root)
+	if diff := deep.Equal(tree3, []string{""}); diff != nil {
+		t.Fatalf("Found diff after remove: %s", diff)
+	}
+}
+
+func dumpPathTree(t testing.TB, root string) []string {
```

Comment:
Not sure what this function does.  I think it works well (only) when the passed-in directory contains only zero or one elements, recursively all the way down.  Otherwise `Walk` does not guarantee the order in which files are passed to the callback.
Doesn't matter in *this* test, I think, but it's worthy of at least documentation.  Or if you want to fix, consider collecting entire `TrimPrefix(path, root)`s and then sorting.

Reply:
I will keep the result sorted - but I must remove the prefix as it is a temporary folder.